spec/features/generic_announcements_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe 'Generic announcements', type: :feature do
+  subject { page }
+  let(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+    let(:user) { create(:user) }
+
+    describe 'no generic announcements' do
+      before do
+        instance.announcements.clear
+        SystemAnnouncement.destroy_all
+        visit announcements_path
+      end
+
+      it { is_expected.not_to have_selector('div.instance-announcement') }
+      it { is_expected.not_to have_selector('div.system-announcement') }
+    end
+
+    describe 'one instance announcement' do
+      let(:announcement) do
+        build(:instance_announcement, instance: instance, creator: user, updater: user)
+      end
+
+      before do
+        instance.announcements.clear
+        SystemAnnouncement.destroy_all
+        announcement.save
+        visit announcements_path
+      end
+
+      it 'shows the announcement' do
+        it { is_expected.to have_selector('div.instance-announcement') }
+      end
+    end
+
+    describe 'one system announcement' do
+      let(:announcement) do
+        build(:system_announcement, creator: user, updater: user)
+      end
+
+      before do
+        instance.announcements.clear
+        SystemAnnouncement.destroy_all
+        announcement.save!
+        visit announcements_path
+      end
+
+      it 'shows the announcement' do
+        it { is_expected.to have_selector('div.system-announcement') }
+      end
+    end
+
+    describe 'many announcements' do
+      let(:announcements) do
+        [].tap do |result|
+          5.times do
+            result << build(:system_announcement, creator: user, updater: user)
+            result << build(:instance_announcement, instance: instance, creator: user, updater: user)
+          end
+        end
+      end
+
+      before do
+        instance.announcements.clear
+        SystemAnnouncement.destroy_all
+        announcements.each do |announcement|
+          announcement.save!
+        end
+        visit announcements_path
+      end
+
+      it 'shows all announcements' do
+        5.times do |i|
```

Comment:
Unused block argument - `i`. You can omit the argument if you don't care about it.